src/univariate/continuous/chernoff.jl

```diff
@@ -0,0 +1,311 @@
+```
+"""
+struct Chernoff <: ContinuousUnivariateDistribution
+end
+
+# The following arrays of constants have been precomputed to speed up computation.
+# The first two correspond to the arrays a and b in the Groeneboom and Wellner artcle.
+# The array az contains roots of the airyai functions (atilde in the paper).
+# Finally azp contains contains airyaiprime evaluated at the corresponding atildes
+
+a=[
+     0.14583333333333334
+    -0.0014105902777777765
+    -2.045269786155239e-5
+     1.7621771897199738e-6
+    -5.7227597511834636e-8
+     1.2738111577329235e-9
+    -2.1640824986157608e-11
+     2.87925759635786e-13
+    -2.904496582565578e-15
+     1.8213982724416428e-17
+     4.318809086319947e-20
+    -3.459634830348895e-21
+     6.541295360751684e-23
+    -8.92275524974091e-25
+     1.0102911018761945e-26
+    -9.976759882474874e-29
+    ]
+b=[
+     0.6666666666666666
+     0.021164021164021163
+    -0.0007893341226674574
+     1.9520978781876193e-5
+    -3.184888134149933e-7
+     2.4964953625552273e-9
+     3.6216439304006735e-11
+    -1.874438299727797e-12
+     4.393984966701305e-14
+    -7.57237085924526e-16
+     1.063018673013022e-17
+    -1.26451831871265e-19
+     1.2954000902764143e-21
+    -1.1437790369170514e-23
+     8.543413293195206e-26
+    -5.05879944592705e-28
+    ]
+az=[
+    -2.338107410459767
+    -4.08794944413097
+    -5.520559828095551
+    -6.786708090071759
+    -7.944133587120853
+    -9.02265085334098
+    -10.040174341558085
+    -11.008524303733262
+    -11.936015563236262
+    -12.828776752865757
+    -13.691489035210719
+    -14.527829951775335
+    -15.340755135977997
+    -16.132685156945772
+    -16.90563399742994
+    -17.66130010569706
+    ]
+azp=[
+     0.7012108227206906
+    -0.8031113696548534
+     0.865204025894141
+    -0.9108507370495821
+     0.9473357094415571
+    -0.9779228085695176
+     1.0043701226603126
+    -1.0277386888207862
+     1.0487206485881893
+    -1.0677938591574279
+     1.0853028313507
+    -1.1015045702774968
+     1.1165961779326556
+    -1.1307323104931881
+     1.1440366732735523
+    -1.156609849116565
+    ]
+
+cuberoottwo = cbrt(2.0)
+sqrthalfpi = sqrt(0.5*pi)
+sqrttwopi = sqrt(2.0*pi)
+
+function p(y::Real)
+    if iszero(y)
+        return -sqrt(0.5*pi) 
+    end
+    
+    (y>0) || throw(DomainError(y, "y must be positive"))
+    
+    cnsty = y^(-1.5)
+    return (y<=1.0) ? sum([(b[k]*cnsty-a[k]*sqrthalfpi)*y^(3*k) for k=1:length(a)])-sqrthalfpi :
+                      sum([exp(cuberoottwo*az[k]*y) for k=1:length(az)]) * 2 * sqrttwopi* exp(-y*y*y/6) - cnsty
+end
+
+function g(x::Real)
+    function gone(y::Real)
+        return p(y) * exp(-0.5*y*(2*x+y)*(2*x+y))
+    end
+    function gtwo(y::Real)
+        z=2*x+y*y
+        return (z*y*y +0.5 * z*z) * exp(-0.5*y*y*z*z)
+    end
+    return (x <= -1.0) ? cuberoottwo*cuberoottwo * exp(2*x*x*x/3.0) * sum([exp(-cuberoottwo*az[k]*x) / azp[k] for k=1:length(az)]) :
```

Comment:
split on two lines with regular if (this line is a bit too long and dense)

Reply:
also, `a in az` since you're not using `k` for other things than indexing in `az`